stored_requests/backends/db_provider/mysql_dbprovider.go

```diff
@@ -75,7 +86,92 @@ func (provider *MySqlDbProvider) ConnString() string {
 		buffer.WriteString(provider.cfg.Database)
 	}
 
-	return buffer.String()
+	// TLS connection
+	var tls = "false"
+
+	if provider.cfg.TLS.RootCert != "" {
+		tls = "true"
+
+		if err := setupTLSConfig(provider); err != nil {
+			return "", err
+		}
+
+		if provider.cfg.TLS.ClientCert != "" && provider.cfg.TLS.ClientKey != "" {
+			tls = customTLSKey
+		}
+	}
+
+	if tls != "false" && !strings.Contains(provider.cfg.QueryString, "tls=") {
+		buffer.WriteString("?tls=")
+		buffer.WriteString(tls)
+		// Rest of connection string parameters passed through query_string
+		if provider.cfg.QueryString != "" {
+			buffer.WriteString("&")
+			buffer.WriteString(provider.cfg.QueryString)
+		}
+	}
+
+	// Rest of connection string parameters passed through query_string
+	if provider.cfg.QueryString != "" {
+		buffer.WriteString("?")
+		buffer.WriteString(provider.cfg.QueryString)
+	}
```

Comment:
Do we even need this  `tls=prebid-tls` parameter? It's a hardcoded value that adds complications to assembling the connection string. 
Would this work instead: 

stored_request.go: make TLS a pointer. It's simpler to check if it's nil: 
```
TLS         *TLS   `mapstructure:"tls"`
```
And then just:
```
tlss := provider.cfg.TLS != nil
	if tlss {
		if err := setupTLSConfig(provider); err != nil {
			return "", err
		}
	}
if provider.cfg.QueryString != "" {
		buffer.WriteString("?")
		buffer.WriteString(provider.cfg.QueryString)
	}
```
Do I miss something?

Reply:
I tried the suggested approach and it didn't work without passing the same custom tls key that used to call `mysql.RegisterTLSConfig` method.
Unfortunately, that doesn't work because of the way how the mysql driver configures a secure connection:
https://pkg.go.dev/github.com/go-sql-driver/mysql#section-readme

![image](https://user-images.githubusercontent.com/32890592/220600519-4e7dd38d-16e8-40a7-91c2-f2c94b62e78c.png)

![image](https://user-images.githubusercontent.com/32890592/220600048-b18c456a-8a1e-4b77-960c-44da2605948c.png)

Reply:
Oh I see. Ok, I'll test it locally next time.

Reply:
I see why now! If this line was executed `mysql.RegisterTLSConfig(customTLSKey, &tls.Config{` then in connection string tls **must** be equal to `customTLSKey`. 
With current code if only `RootCert` is specified, then `tls=true` goes to connection string and it fails to connect to DB.
I tested this locally, it works with/without QueryString and with/without TLS:

```
func (provider *MySqlDbProvider) ConnString() (string, error) {
	buffer := bytes.NewBuffer(nil)

	if provider.cfg.Username != "" {
		buffer.WriteString(provider.cfg.Username)
		if provider.cfg.Password != "" {
			buffer.WriteString(":")
			buffer.WriteString(provider.cfg.Password)
		}
		buffer.WriteString("@")
	}

	buffer.WriteString("tcp(")
	if provider.cfg.Host != "" {
		buffer.WriteString(provider.cfg.Host)
	}

	if provider.cfg.Port > 0 {
		buffer.WriteString(":")
		buffer.WriteString(strconv.Itoa(provider.cfg.Port))
	}
	buffer.WriteString(")")

	buffer.WriteString("/")

	if provider.cfg.Database != "" {
		buffer.WriteString(provider.cfg.Database)
	}

	if provider.cfg.QueryString != "" {
		buffer.WriteString("?")
		buffer.WriteString(provider.cfg.QueryString)
	}

	if provider.cfg.TLS != nil {

		if err := setupTLSConfig(provider); err != nil {
			return "", err
		}
		tlsParam := "tls=" + customTLSKey

		addTls := true

		if provider.cfg.QueryString != "" {
			if strings.Contains(provider.cfg.QueryString, "tls=") {
				if strings.Contains(provider.cfg.QueryString, tlsParam) {
					//query string has correct tls param
					addTls = false
				} else {
					//error - tls name doesn't match
					return "", errors.New("Incorrect query string")
				}
			} else {
				//ad tls after query string
				buffer.WriteString("&")
				buffer.WriteString(tlsParam)
				addTls = false
			}
		}
		if addTls {
			//no query string, add tls only
			buffer.WriteString("?")
			buffer.WriteString(tlsParam)
		}
	}

	return buffer.String(), nil
}

```